snippet_parser/core.py

```diff
@@ -113,6 +117,100 @@ def _fast_parse(self, wikitext):
         except mwparserfromhell.parser.ParserError:
             return None
 
+    def extract_from_sentences(self, wikitext, sentences):
+        for sent in sentences:
+            start = wikitext.find(sent)
+            marked = _SENTENCE_MARKER_MARKUP.format(sent = sent)
+            if start >= 0:
+                end = start + len(sent)
+                wikitext = wikitext[:start] + marked + wikitext[end:]
+            else:
+                print("Not found: ", sent)
+
+        wikicode = mwparserfromhell.parse(wikitext)
```

Comment:
From here on, there's a lot of duplication between this method and the regular `extract`. That is fine for now while you try out this approach, but before we merge this code, we'll want to reduce the duplicated code a little by possibly breaking up the current `extract` into a few more methods that can be called from both `extract` and `extract_sentences`.

Reply:
Alright, I see!